Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialCanvasDocument.cpp

```diff
@@ -352,4 +372,337 @@ namespace MaterialCanvas
+        };
+
+        AZStd::vector<GraphModel::ConstNodePtr> nodes = GetNodesInExecutionOrder();
+
+        for (const auto& node : nodes)
+        {
+            if (auto dynamicNode = azrtti_cast<const AtomToolsFramework::DynamicNode*>(node.get()))
+            {
+                const auto& nodeConfig = dynamicNode->GetConfig();
+
+                // Prescan to see if this node has any output connections
+                // Generally speaking, we can ignore the node if there are no output connections but we're just trying to gather data now.
+                bool hasOutputConnections = false;
+                for (const auto& slotConfig : nodeConfig.m_outputSlots)
+                {
+                    auto slot = dynamicNode->GetSlot(slotConfig.m_name);
+                    if (slot && !slot->GetConnections().empty())
+                    {
+                        hasOutputConnections = true;
+                        break;
+                    }
+                }
+
+                collectCommonSettings(nodeConfig.m_settings);
+
+                for (const auto& slotConfig : nodeConfig.m_propertySlots)
+                {
+                    collectCommonSettings(slotConfig.m_settings);
+
+                    // Write out any property slot data if there were any connections
+                    if (hasOutputConnections)
+                    {
+                        auto slot = dynamicNode->GetSlot(slotConfig.m_name);
+                        AZStd::vector<AZStd::string> instructionsForSlot;
+                        collectSettingsAsVec(slotConfig.m_settings, "instructions", instructionsForSlot);
+                        updateAndAddSlotInstructions(node, slot, instructionsForSlot);
+                    }
+                }
+
+                for (const auto& slotConfig : nodeConfig.m_inputSlots)
+                {
+                    collectCommonSettings(slotConfig.m_settings);
+
+                    // For now, this will always generate all input slot instructions but can potentially be optimized to eliminate
+                    // instructions that result in temporary variables that are copies would pass through two other variables.
+                    auto slot = dynamicNode->GetSlot(slotConfig.m_name);
+                    AZStd::vector<AZStd::string> instructionsForSlot;
+                    collectSettingsAsVec(slotConfig.m_settings, "instructions", instructionsForSlot);
+
+                    // Input slots will replace the value with the name of the variable from the connected help with slot if one is set.
+                    if (slot && !slot->GetConnections().empty())
+                    {
+                        for (const auto& connection : slot->GetConnections())
+                        {
+                            auto sourceSlot = connection->GetSourceSlot();
+                            if (sourceSlot && sourceSlot->GetSlotDirection() == GraphModel::SlotDirection::Output)
```

Comment:
Since we're looping over `m_inputSlots`, in what situations is this slot direction actually incorrect?

Reply:
I was being explicit just in case. I need to talk to @cgalvan to confirm that the source slot referenced in the connection will always be and output slot and it's not based on where the drag started to create the connection.

Reply:
@gadams3 I can confirm regardless of which slot you originated the drag from, the source will always be an output direction and the target will always be input.